libbeat/publisher/pipeline/module.go

```diff
@@ -170,32 +172,54 @@ func loadOutput(
 	return out, nil
 }
 
-func createQueueBuilder(
+func createQueueFactory(
 	config conf.Namespace,
 	monitors Monitors,
 	inQueueSize int,
-) (func(queue.ACKListener) (queue.Queue, error), error) {
+) (queueFactory, error) {
 	queueType := defaultQueueType
 	if b := config.Name(); b != "" {
 		queueType = b
 	}
 
-	queueFactory := queue.FindFactory(queueType)
-	if queueFactory == nil {
-		return nil, fmt.Errorf("'%v' is no valid queue type", queueType)
+	if monitors.Telemetry != nil {
+		queueReg := monitors.Telemetry.NewRegistry("queue")
+		monitoring.NewString(queueReg, "name").Set(queueType)
 	}
 
-	queueConfig := config.Config()
-	if queueConfig == nil {
-		queueConfig = conf.NewConfig()
+	switch queueType {
+	case "mem":
```

Comment:
before the queue type was coming from RegisterQueueType in the directory with the queue definition, now it is far away in pipeline directory.  what about exporting a QueueType variable and using that in the switch statement?  I'm open to other suggestions too.  I just don't like the magic string for what determines the queue type being far away from the queue definitions.

Reply:
Fair enough, I made these strings exported constants from the queue packages